src/MongoClient.php

```diff
@@ -376,6 +376,8 @@ public function insert(string $collection, array $document, array $options = [])
             $docObj->{$key} = $value;
         }
 
+        $docObj->_id ??= new BSON\ObjectId();
```

Comment:
Looks very nice. Is it a unique a id in terms of race condition?
Then to assign to collection and remove lastInsertedDocument?